python/semantic_kernel/planning/plan.py

```diff
@@ -381,7 +391,7 @@ def get_next_step_variables(
         # - Step Parameters (pull from variables or state by a key value)
         function_params = step.describe()
         for param in function_params._parameters:
-            if param.name.lower == "input":
+            if param.name.lower() == "input":
```

Comment:
great catch here. This hid bugs in string checking, so I'm pushing some changes to resolve and fix the unit tests. 

There are several areas where I changed the pattern like below:
```
-        if input is not None:
+        if input is not None and input != "":
```